lib/api/apiUtils/bucket/models/ReplicationConfiguration.js

```diff
@@ -122,12 +122,8 @@ class ReplicationConfiguration {
         if (!Role) {
             return errors.MalformedXML;
         }
-        // TODO: Update to validate role priveleges after implemented in Vault.
-        // Role should be an IAM user name.
-        const arr = Role[0].split(':');
-        const isValidARN = arr.length === 7 ||
-            (arr.length === 6 && arr[5].split('/').length === 2);
-        if (!isValidARN) {
+        const regexp = /^arn:aws:iam::.*:role.*/;
```

Comment:
detail: I don't know exactly what is a valid ARN, but you may want to replace `.*` before `:role` with `[^:]*` to exclude the colon character, or even only allow the subset of characters allowed in the ARN path item?